app/src/main/java/org/simple/clinic/widgets/ItemAdapter.kt

```diff
@@ -0,0 +1,40 @@
+package org.simple.clinic.widgets
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.annotation.LayoutRes
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import io.reactivex.Observable
+import io.reactivex.subjects.PublishSubject
+import io.reactivex.subjects.Subject
+import org.simple.clinic.widgets.recyclerview.ViewHolderX
+
+open class ItemAdapter<I : ItemAdapter.Item<E>, E>(
+    diffCallback: DiffUtil.ItemCallback<I>
+) : ListAdapter<I, ViewHolderX>(diffCallback) {
+
+  private val eventSubject: Subject<E> = PublishSubject.create<E>()
+
+  val listItemEvents: Observable<E> = eventSubject.hide()
```

Comment:
Rename this to `itemEvents`?
Since we are not using the term `list` anywhere in this class. Its _obvious_ :)